### DISCUSSION TITLE
Nested <View /> to clip contents

### DISCUSSION THREAD

**Discussion by kkirby** [pmndrs/drei#905] [Q&A]
Does anyone know of a way to nest views? I know how I'd go about implementing it myself, but I'd rather not re-invent the wheel if it's already been created.

The issue I have is that I have a scrolling container, and I want the contents to clip the to that container. Here's a video showcasing the issue:

Expected:

![CleanShot 2022-05-16 at 08 08 22](https://user-images.githubusercontent.com/34333/168599649-be3643bd-d1eb-4d34-9e7a-fbbcea0099f1.gif)

Actual:

![CleanShot 2022-05-16 at 08 05 34](https://user-images.githubusercontent.com/34333/168599286-a1e40863-0c29-4231-8fa0-f96ec0ac87ef.gif)

Like I said, I know how I could go about doing this myself, but if it's already been solved, I've love to use that.

My solution would be to use framebuffers for subviews and then render them relative to the parent view. The parent view then does the clipping.

Thanks!

**Comment by drcmda** [comment]
the premise is that views are controlled by divs, nothing inside leaks out but to clip it wouldnt be easy.

probably just use this instead https://github.com/pmndrs/drei#mask

**Reply by kkirby** [reply]
Yep, that would 100% solve the issue. Thanks!

**Reply by kkirby** [reply]
It seems like Masks and Views don't really want to work together. If I define a mask outside a View, the coordinate space is different, so the content isn't masked correctly.
